### PR TITLE
TW-759 Use keyboard to select suggestion item

### PR DESCRIPTION
**Pull request by drminh2807**
# Issue
- #759 

# Demo

https://github.com/linagora/twake-on-matrix/assets/12546908/a181c8c9-41a1-4028-acd5-34b02a7369aa

https://github.com/linagora/twake-on-matrix/assets/12546908/6b4c33f6-d172-48ef-b69a-a1e087081b01

### CODE REVIEW

File: lib/pages/chat/input_bar/focus_suggestion_controller.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  List<Map<String, String?>> _suggestions = List.empty();
+
+  List<Map<String, String?>> get suggestions => _suggestions;
+
+  set suggestions(List<Map<String, String?>> suggestions) {
+    _suggestions = suggestions;
+    currentIndex.value = 0;
+  }
+
+  final currentIndex = ValueNotifier(0);
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/focus_suggestion_controller.dart:13]: don't forget to dispose when not use

File: lib/pages/chat_draft/draft_chat.dart [hunk 2]
```diff
@@ -52,6 +53,9 @@ class DraftChatController extends State<DraftChat>
   final AutoScrollController scrollController = AutoScrollController();
   final AutoScrollController forwardListController = AutoScrollController();
 
+  final FocusSuggestionController focusSuggestionController =
+      FocusSuggestionController();
```
Comment (nqhhdev) [inline lib/pages/chat_draft/draft_chat.dart:57]: Same above

File: lib/pages/chat/input_bar/focus_suggestion_controller.dart (hunk 1 above)
Comment (drminh2807) [reply lib/pages/chat/input_bar/focus_suggestion_controller.dart:13]: done

File: lib/pages/chat_draft/draft_chat.dart (hunk 2 above)
Comment (drminh2807) [reply lib/pages/chat_draft/draft_chat.dart:57]: done

**Review by hoangdat — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

File: lib/pages/chat/input_bar/focus_suggestion_list.dart [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    widget.scrollController.animateTo(
+      offset,
+      duration: _scrollAnimationDuration,
+      curve: Curves.linear,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final listHeight = InputBarStyle.suggestionSize *
```
Comment (sherlockvn) [inline lib/pages/chat/input_bar/focus_suggestion_list.dart:53]: should put in style

**Comment by DessisInformatica** [comment]
@drminh2807 Could you share the code used to function the arrows inside the TypeAheadField?

**Pull request merged**